### PR TITLE
Edit username in Account Settings

### PR DESCRIPTION
**Pull request by juanmanzojr**
Allows users to edit their username within the Accounts Settings page. Adds a validation for `username` which checks if the username contains profanity.
Screenshot 2024-02-01 at 10 47 37 AM
Screenshot 2024-02-01 at 11 46 03 AM
Screenshot 2024-02-06 at 2 46 39 PM

## Links

- jira ticket: [here](https://codedotorg.atlassian.net/browse/P20-374)
Note: Mock ups in Jira contain the new styling which will be implemented in a different PR:
Slack thread for styling changes: [here](https://codedotorg.slack.com/archives/CFTFD6BPV/p1706810549232389)

## Follow-up work
Profanity check for usernames and display names.
Styling changes

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

**Comment by juanmanzojr** [comment]
This PR is a continuation of #56199, which was mis-closed by the Git LFS migration (#55759).

### Previous Comments:


### Previous Reviews:
- **carl-codeorg changes_requested** - 
- **carl-codeorg approved** - LGTM! Did we do a quick sanity check into WebPurify costs, just to make sure the change won't cause us to hit any limits and estimate costs?

File: dashboard/app/views/devise/registrations/edit.html.haml [hunk 1]
```diff
@@ -44,7 +44,7 @@
   - if resource.username.present?
     .field
       = f.label :username, class: "label-bold"
-      = f.object.username
+      = f.text_field :username, autofocus: false, maxlength: 20
```
Comment (carl-codeorg) [inline dashboard/app/views/devise/registrations/edit.html.haml]: Can we also do a minlength of 5 on the field? The user model has `validates_length_of :username, within: 5..20, allow_blank: true`

**Comment by carl-codeorg** [comment]
The [USERNAME_REGEX validation](https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/app/models/user.rb#L536) is only running on `:create`. Should probably run that on update as well to avoid unwanted characters in the new name,

**Review by carl-codeorg — APPROVED** [review]
LGTM

**Pull request merged**